pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1374,6 +1378,14 @@ private void assignInstances(TableConfig tableConfig, boolean override) {
    */
   public void updateTableConfig(TableConfig tableConfig)
       throws IOException {
+    TenantConfig tenantConfig = tableConfig.getTenantConfig();
+    String brokerTag = tenantConfig.getBroker();
+    String serverTag = tenantConfig.getServer();
+    if (brokerTag == null || serverTag == null) {
+      if (!_isSingleTenantCluster) {
+        throw new InvalidTableConfigException("server and broker tenants must be specified for multi-tenant cluster");
+      }
+    }
```

Comment:
You should put default tenant here if it is single-tenant cluster, similar to `addTable()`

You may extract this part into a helper method

Reply:
makes sense. lemme do that